.mergify.yml

```diff
@@ -0,0 +1,27 @@
+queue_rules:
+  - name: default
+    conditions: []
+
+pull_request_rules:
+  - name: Automatic merge on approval
+    conditions:
+      - "base=main"
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - "check-success=buildkite/pull-request"
+      - "check-success=buildkite/pull-request/end-to-end-test"
+      - "check-success=buildkite/pull-request/end-to-end-test-parallel"
+      - "check-success=buildkite/docker"
+      - "check-success=buildkite/pull-request/build"
+      - "check-success=buildkite/pull-request/end-to-end-source-test"
+      - "check-success=buildkite/pull-request/misc-check"
+      - "check-success=buildkite/pull-request/pipeline-pipeline-upload"
+      - "check-success=buildkite/pull-request/pre-unit-test"
+      - "check-success=buildkite/pull-request/unit-test"
+      - "check-success=buildkite/pull-request/unit-test-deterministic-simulation"
```

Comment:
How about adding a check `buildkite/finish` as the endpoint of the DAG (depends on all other checks) and always success, so that we can write only a check here.

Reply:
Maybe only the first-level directory here. The first-level directory represents the pipeline. Only if all the steps in the pipeline pass, the pipeline will pass.

Reply:
like:
```
- "check-success=buildkite/docker"
- "check-success=buildkite/pull-request"
```

Reply:
looks good!